bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/internal/i18n/I18nConfigOptionsProvider.java

```diff
@@ -23,29 +28,65 @@
  * {@link ConfigOptionProvider} that provides a list of
  *
  * @author Simon Kaufmann - initial contribution and API
+ * @author Erdoan Hadzhiyusein - Added time zone
  *
  */
 @Component(immediate = true)
 public class I18nConfigOptionsProvider implements ConfigOptionProvider {
 
+    private static final String NO_OFFSET_FORMAT = "(GMT) %s";
+    private static final String NEGATIVE_OFFSET_FORMAT = "%s (GMT%d:%02d)";
+    private static final String POSITIVE_OFFSET_FORMAT = "%s (GMT+%d:%02d)";
+
     @Override
     public Collection<ParameterOption> getParameterOptions(URI uri, String param, Locale locale) {
         if (uri.toString().equals("system:i18n")) {
             Locale translation = locale != null ? locale : Locale.getDefault();
-            if (param.equals("language")) {
-                return getAvailable(locale, l -> new ParameterOption(l.getLanguage(), l.getDisplayLanguage(translation)));
-            } else if (param.equals("region")) {
-                return getAvailable(locale, l -> new ParameterOption(l.getCountry(), l.getDisplayCountry(translation)));
-            } else if (param.equals("variant")) {
-                return getAvailable(locale, l -> new ParameterOption(l.getVariant(), l.getDisplayVariant(translation)));
+            switch (param) {
+                case "language":
+                    return getAvailable(locale,
+                            l -> new ParameterOption(l.getLanguage(), l.getDisplayLanguage(translation)));
+                case "region":
+                    return getAvailable(locale,
+                            l -> new ParameterOption(l.getCountry(), l.getDisplayCountry(translation)));
+                case "variant":
+                    return getAvailable(locale,
+                            l -> new ParameterOption(l.getVariant(), l.getDisplayVariant(translation)));
+                case "timezone":
+                    Collection<ParameterOption> collection = new LinkedList<ParameterOption>();
+                    String[] ids = TimeZone.getAvailableIDs();
```

Comment:
I assume you are more familar with the topic of Java 8's new time API because you implemented https://github.com/eclipse/smarthome/pull/4259

But shouldn't we use `ZoneId` instead of `TimeZone`?

```
There is an existing time zone class in Java—java.util.TimeZone—but it isn’t used by Java SE 8 be-cause all JSR 310 classes are immutable and time zone is mutable.
```

There is also `ZoneId.getAvailableZoneIds()`, ...